app/src/main/java/org/thoughtcrime/securesms/onboarding/SeedActivity.kt

```diff
@@ -29,10 +35,10 @@ class SeedActivity : BaseActionBarActivity() {
         if (hexEncodedSeed == null) {
             hexEncodedSeed = IdentityKeyUtil.getIdentityKeyPair(this).hexEncodedPrivateKey // Legacy account
         }
-        val loadFileContents: (String) -> String = { fileName ->
-            MnemonicUtilities.loadFileContents(this, fileName)
+
+        runBlocking {
```

Comment:
runblocking in prodiction code!
I'm pretty sure we should never have runBlocking in an android app, right?
Especially not constrained to a specific dispatcher...
Is there a reason not to use the lifecycleScope?
Are there more runBlocking usage in the app?

Reply:
Is it because you want the task to outlive the activity's lifecycle?

Reply:
It's good you call it out, this is the exact effect I'd expect to see: anti-pattern needs to be screamed at! Just like the philosophy behind the `!!` operator in kotlin. 

The reason I put it there is previously during one of the `MnemonicUtilities` chain of calls, there is one function that accesses the IO system. This utility is used on main thread. I changed that particular IO access to a suspend function. However, due to the suspend function being used, all the callers will then need to be suspend function. At some point, the change over to suspend function became non-trivial and I don't want to do too much refactoring in one PR. 

Hence I decided the use of `runBlocking` + `suspend func` is no worse than doing blocking IO on main thread.

Reply:
But why can't the lifecycleScope not be used here?
And why not `runBlocking(Dispatchers.IO)` ? Is the content doing things that require the main thread?

Reply:
Yes if you look at `SeedActivity`, `seed` is a lazy variable the main thread needs to access immediately. I couldn't just change it to use suspend function, because it requires rewrite of the whole activity.

I'm not sure how `runBlocking(Dispatchers.IO)` makes any difference in this case? And how is `lifecycleScope` going to help here?

I totally understand the POV that "why adding more tech debt while fixing a debt". However I'd argue if I'm actually adding tech debt: what I do is reducing tech debt in other places while putting a mark on the existing tech debt here without fixing it, it's still an overall debt reduction.

Reply:
Well can't you call `lifecycleScope.launch{` instead of `runBlocking`?
As for the Dispatcher type, it's to avoid using the default thread, which will be the main thread, unless the internal work in there interacts with the main thread, in which case you can leave the default.

Reply:
Before the change the code is calling `MnemonicCodec(loadFileContents).encode(hexEncodedSeed!!, MnemonicCodec.Language.Configuration.english)` .

This is returns a string as a regular function and this return value is used immediately. I have updated `encode` to a suspend function, and to be able to use the return value immediately, you'd have to block on the suspend function. If you use `launch` you can't really get the value right away.

Reply:
I did move this into the VM in onboarding...

``` 
val seed by lazy {
        val hexEncodedSeed = IdentityKeyUtil.retrieve(application, IdentityKeyUtil.LOKI_SEED)
            ?: IdentityKeyUtil.getIdentityKeyPair(application).hexEncodedPrivateKey // Legacy account
        MnemonicCodec { MnemonicUtilities.loadFileContents(application, it) }
            .encode(hexEncodedSeed, MnemonicCodec.Language.Configuration.english)
}
```

https://github.com/oxen-io/session-android/blob/5459b1eb1d5c2f873fea30125e0477186cde01ec/app/src/main/java/org/thoughtcrime/securesms/onboarding/recoverypassword/RecoveryPasswordViewModel.kt#L39-L44

Reply:
but yeah, still on main thread I think

Reply:
Fine with leaving it if you guys are happy with it this way. Could we maybe add a //todo or //fixme on both the runBlocking as a reminder to deal with it once we properly apply suspend functions and coroutines everywhere?